pkg/paho-mqtt/contrib/riot_iface.c

```diff
@@ -79,23 +82,25 @@ static int mqtt_read(struct Network *n, unsigned char *buf, int len,
             rc = 0;
         }
 
-        if (IS_USED(MODULE_LWIP)) {
-            if (rc > 0) {
-                tsrb_add(&tsrb_lwip_tcp, _temp_buf, rc);
-            }
-
-            rc = tsrb_get(&tsrb_lwip_tcp, buf, len);
+#if defined(MODULE_LWIP)
+        if (rc > 0) {
+            tsrb_add(&tsrb_lwip_tcp, _temp_buf, rc);
         }
+
+        rc = tsrb_get(&tsrb_lwip_tcp, buf, len);
+#endif
```

Comment:
Again, I believe this could be left at C level without downsides. (More instances below.)